IdentityServer.LdapExtension/UserStore/InMemoryUserStore.cs

```diff
@@ -55,6 +55,38 @@ public IAppUser ValidateCredentials(string username, string password)
             return default(TUser);
         }
 
+        /// <summary>
+        /// Validates the credentials.
+        /// </summary>
+        /// <param name="username">The username.</param>
+        /// <param name="password">The password.</param>
+        /// <param name="domain">The domain friendly name.</param>
+        /// <returns>
+        /// Returns the application user that match that account if the
+        /// authentication is successful.
+        /// </returns>
+        public IAppUser ValidateCredentials(string username, string password, string domain)
+        {
+            try
+            {
+                var user = _authenticationService.Login(username, password, domain);
+                if (user != null)
+                {
+                    return user;
+                }
+            }
+            catch (LoginFailedException)
+            {
+                return default(TUser);
+            }
+            catch (Exception)

```

Comment:
Don't catch and re-throw. If you don't catch it will automatically throws, don't worry.